thirdparty/etcpak/System.cpp

```diff
@@ -62,7 +65,8 @@ void System::SetThreadName( std::thread& thread, const char* name )
     __except(EXCEPTION_EXECUTE_HANDLER)
     {
     }
-#elif !defined(__APPLE__)
-    pthread_setname_np( thread.native_handle(), name );
+#elif defined __linux__ || defined __MINGW32__
+    pthread_setname_np( (pthread_t)thread.native_handle(), name );
```

Comment:
I added the cast to solve this issue with `llvm-mingw`:
```
x86_64-w64-mingw32-clang++ -o thirdparty/etcpak/System.windows.opt.tools.64.o -c -std=gnu++17 -w -g2 -O2 -O2 -mwindows -Wall -Werror=return-type -w -DDEBUG_ENABLED -DWINDOWS_SUBSYSTEM_CONSOLE -DWINDOWS_ENABLED -DWASAPI_ENABLED -DWINMIDI_ENABLED -DWINVER=0x0601 -D_WIN32_WINNT=0x0601 -DVULKAN_ENABLED -DMINGW_ENABLED -DMINGW_HAS_SECURE_API=1 -DTOOLS_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -Ithirdparty/libpng -Ithirdparty/etcpak -Ithirdparty/libpng -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/vulkan/loader -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/windows -I. thirdparty/etcpak/System.cpp
thirdparty/etcpak/System.cpp:69:5: error: no matching function for call to 'pthread_setname_np'
    pthread_setname_np( thread.native_handle(), name );
    ^~~~~~~~~~~~~~~~~~
/home/akien/Projects/godot/toolchains/llvm-mingw-20201020-ucrt-ubuntu-18.04/x86_64-w64-mingw32/include/pthread.h:317:26: note: candidate function not viable: cannot convert argument of incomplete type 'std::__1::thread::native_handle_type' (aka 'void *') to 'pthread_t' (aka 'unsigned long long') for 1st argument
int       WINPTHREAD_API pthread_setname_np(pthread_t thread, const char *name);
                         ^
1 error generated.
```

I'm not sure it's the proper fix, does that actually imply that the native handle I'm getting is not a pthread handle? In which case I'd be casting to an incompatible type I guess.

It works without the cast with mingw-gcc (at least when using winpthreads).

Reply:
Yeah we debugged with HP, the cast is wrong. I'll push a better fix later.